sklearn/datasets/openml.py

```diff
@@ -63,7 +86,19 @@ def _open_openml_url(openml_path, data_home):
         try:
             with gzip.GzipFile(local_path, 'wb') as fdst:
                 fsrc = urlopen(_OPENML_PREFIX + openml_path)
-                shutil.copyfileobj(fsrc, fdst)
+                if md5_checksum is not None:
+                    md5 = hashlib.md5()
+                    block_size = 128 * md5.block_size
+                    for block in iter(lambda: fsrc.read(block_size), b''):
+                        md5.update(block)
+                        fdst.write(block)
+
+                    if md5_checksum != md5.hexdigest():
+                        warn('Data set file hash {} does not match the '
```

Comment:
Why do we want a warning rather than an exception?

Reply:
I thought an exception would be raised when decoding in `_arff`, but it might be better to fail earlier. Which exception would you raise instead?

Reply:
If it's incomplete but stops at a line break, there will be no exception. We should raise one here